test/task_bulk_complete_test.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from helpers import unittest
+from luigi import Task
+from luigi import Parameter
+from luigi.task import MixinNaiveBulkComplete
+
+COMPLETE_TASKS = ["A", "B", "C"]
+
+
+class MockTask(MixinNaiveBulkComplete, Task):
+    param_a = Parameter()
+    param_b = Parameter("Not Mandatory")
```

Comment:
Nono, I just meant  change the line to `Parameter(default="Not Mandatory"). Because it's more readable over plain `Parameter("Not Mandatory")`. :)

> Is the reason for this that the order of parameters based on initiation order is not guaranteed to be consistent?

It's guaranteed to be consistent. You have understood it correctly I think, sorry if my previous comment was confusing!